static/src/javascripts/projects/common/modules/support/banner.ts

```diff
@@ -0,0 +1,205 @@
+import { getMvtValue } from 'common/modules/analytics/mvt-cookie';
+import { submitComponentEvent } from 'common/modules/commercial/acquisitions-ophan';
+import { getVisitCount } from 'common/modules/commercial/contributions-utilities';
+import { shouldHideSupportMessaging } from 'common/modules/commercial/user-features';
+import {
+	buildTagIds,
+	dynamicImport,
+	getArticleCountConsent,
+	isHosted,
+	ModulesVersion,
+	supportDotcomComponentsUrl,
+	tracking,
+} from 'common/modules/support/supportMessaging';
+import type { Page } from 'common/modules/support/supportMessaging';
+import userPrefs from 'common/modules/user-prefs';
+import config from 'lib/config';
+import fastdom from 'lib/fastdom-promise';
+import { getCountryCode } from 'lib/geolocation';
+import reportError from 'lib/report-error';
+
+export const NO_RR_BANNER_TIMESTAMP_KEY = 'gu.noRRBannerTimestamp'; // timestamp of when we were last told not to show a RR banner
+const twentyMins = 20 * 60000;
+
+export const withinLocalNoBannerCachePeriod = (): boolean => {
+	const item = window.localStorage.getItem(NO_RR_BANNER_TIMESTAMP_KEY);
+	if (item && !Number.isNaN(parseInt(item, 10))) {
+		const withinCachePeriod = parseInt(item, 10) + twentyMins > Date.now();
+		if (!withinCachePeriod) {
+			// Expired
+			window.localStorage.removeItem(NO_RR_BANNER_TIMESTAMP_KEY);
+		}
+		return withinCachePeriod;
+	}
+	return false;
+};
+
+export const setLocalNoBannerCachePeriod = (): void =>
+	window.localStorage.setItem(NO_RR_BANNER_TIMESTAMP_KEY, `${Date.now()}`);
+
+export const renderBanner = (
+	response: ModuleDataResponse,
+): Promise<boolean> => {
+	if (!response.data) {
+		return Promise.resolve(false);
+	}
+	const { module, meta } = response.data;
+
+	return dynamicImport(module.url, module.name)
+		.then((Banner: React.FC<any>) => {
+			const isPuzzlesBanner = module.name === 'PuzzlesBanner';
+
+			return fastdom
+				.mutate(() => {
+					const container = document.createElement('div');
+					container.classList.add('site-message--banner');
+					container.classList.add('remote-banner');
+					if (isPuzzlesBanner) {
+						container.classList.add('remote-banner--puzzles');
+					}
+
+					document.body.insertAdjacentElement('beforeend', container);
+
+					return mountDynamic(
+						container,
+						Banner,
+						{ submitComponentEvent, ...module.props },
+						!isPuzzlesBanner, // The puzzles banner has its own CacheProvider component, and needs this to be false
+					);
+				})
+				.then(() => {
+					// track banner view event in Google Analytics for subscriptions banner
+					if (
+						meta.componentType ===
+						'ACQUISITIONS_SUBSCRIPTIONS_BANNER'
+					) {
+						trackNonClickInteraction(
+							'subscription-banner : display',
+						);
+					}
+
+					return true;
+				});
+		})
+		.catch((error: any) => {
+			/* eslint-disable @typescript-eslint/restrict-template-expressions -- error log */
+			console.log(`Error importing remote banner: ${error}`);
```

Comment:
I'd suggest using the `@guardian/libs` logger here. It pretty prints logs, and only shows it to relevant teams, even in production.

Reply:
oh nice! Though I now need to add supporter revenue to the set of teams

Reply:
updating to use `log`, with a new team name!